types/validator_set.go

```diff
@@ -74,6 +74,9 @@ func (vals *ValidatorSet) CopyIncrementProposerPriority(times int) *ValidatorSet
 // proposer. Panics if validator set is empty.
 // `times` must be positive.
 func (vals *ValidatorSet) IncrementProposerPriority(times int) {
+	if vals.IsNilOrEmpty() {
+		return
```

Comment:
I don't like the fact that we simply return from the function 👎 Note the function's comment says "Panics if validator set is empty.", which is not case! Moreover, if we have a contract where validator set can't be ever empty, we should panic, not silently return. Imagine someone made an error in the code and validator set is empty => we return and continue execution with WRONG DATA. It's OK to panic if the invariants are not satisfied.

If validator set can be empty sometimes, we should use null object pattern https://www.tutorialspoint.com/design_pattern/null_object_pattern.htm